src/pages/index.js

```diff
@@ -64,50 +83,155 @@ profileEditPopup.setEventListeners();
   profileEditPopup.open();
 });
 
 cardAddButton.addEventListener("click", function () {
   newCardPopup.open();
 });
 
+// avatarPopup
+
+const avatarPopup = new PopupWithForm(
+  "#profile-picture-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+avatarEditButton.addEventListener("click", function () {
+  avatarPopup.open();
+});
+
 // Event Handlers
 
 function handleImageClick({ name, link }) {
   imagePreviewPopup.open({ name, link });
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, job: data.description });
-  imagePreviewPopup.close();
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(data)
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, about: res.about });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setLoading(false);
+    });
 }
 
 function handleAddCardFormSubmit(inputValues) {
   const name = inputValues.name;
   const link = inputValues.link;
-  const cardData = { name: name, link: link };
-  renderCard(cardData);
-  newCardPopup.close();
-  cardFormElement.reset();
-  addCardFormValidator.disableButton();
+  newCardPopup.setLoading(true);
+  api
+    .addCard({ name, link })
+    .then((cardData) => {
+      renderCard(cardData);
+      newCardPopup.close();
+      cardFormElement.reset();
+      addCardFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardPopup.setLoading(false);
+    });
+}
+
+function handleAvatarFormSubmit(userData) {
+  avatarPopup.setLoading(true);
+  api
+    .updateUserAvatar(userData)
+    .then((res) => {
+      userInfo.setUserAvatar({ avatar: res.avatar });
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarPopup.setLoading(false);
+    });
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardclick,
+    handleLikeCard,
+    handleUnLikeCard
+  );
   return card.getView();
 }
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
+
+function handleDeleteCardclick(card) {
+  cardDeletePopup.open();
+  cardDeletePopup.setSubmitAction(() => {
+    cardDeletePopup.setLoading(true);
+    api
+      .removeCard(card._id)
+      .then(() => {
+        card._handleDeleteCard();
+        cardDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopup.setLoading(false, "Yes");
+      });
+  });
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+  }
+  api
+    .likeCard(card._id)
+    .then(() => {
+      card._setCardLike(true);
```

Comment:
`_setCardLike`

* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.